lib/validation.js

```diff
@@ -1,40 +1,36 @@
 'use strict'
 
-const fastJsonStringify = require('fast-json-stringify')
-const Ajv = require('ajv')
-
-const bodySchema = Symbol('body-schema')
-const querystringSchema = Symbol('querystring-schema')
+const headersSchema = Symbol('headers-schema')
 const paramsSchema = Symbol('params-schema')
+const querystringSchema = Symbol('querystring-schema')
+const bodySchema = Symbol('body-schema')
+
 const responseSchema = Symbol('response-schema')
-const headersSchema = Symbol('headers-schema')
-const kFluentSchema = Symbol.for('fluent-schema-object')
 
-function getValidatorForStatusCodeSchema (statusCodeDefinition, externalSchema) {
-  return fastJsonStringify(statusCodeDefinition, { schema: externalSchema })
-}
+function compileSchemasForSerialization (context, compile) {
+  if (!context.schema || !context.schema.response) {
+    return
+  }
 
-function getResponseSchema (responseSchemaDefinition, sharedSchemas) {
-  var statusCodes = Object.keys(responseSchemaDefinition)
-  return statusCodes.reduce(function (r, statusCode) {
-    r[statusCode] = getValidatorForStatusCodeSchema(responseSchemaDefinition[statusCode], sharedSchemas)
-    return r
-  }, {})
+  const { method, url } = context.config || {}
+  context[responseSchema] = Object.keys(context.schema.response)
```

Comment:
Why does this function incur side effects?

Reply:
Here the compilers (validation and serialization) are saving to the route's context the compiled functions.

Only this file is writing and reading those functions via `Symbol`s

I keep this logic untouched since changing it would mean a refactor to the `context.js` prototype

Reply:
Done